File: 338. Counting Bits.java
class Solution {
    public int[] countBits(int num) {
        int[] dp = new int[num + 1];

        for(int i = 1; i < num + 1; i ++) {
            dp[i] = dp[i / 2] + (i & 1);
        }
        return dp;
    }
}

File: 338. Counting Bits.cpp
class Solution {
public:
    vector<int> countBits(int num) {
        vector<int> result(num + 1);
        for(int i = 1; i < num + 1; i ++) {
            result[i] = result[i >> 1] + (i & 1);
        }
        return result;
    }
};